### PR TITLE
[native] Pass required subfields to Velox HiveColumnHandle

### PR DESCRIPTION
**Pull request by Yuhta**
Differential Revision: D44342800

The values were currently not passed as Velox did not have support for it.  Now the support for subfields pruning is completed in Velox, we can pass the values and benefit from it.

### CODE REVIEW

**Review by amitkdutta — APPROVED** [review]
Looks good.

File: presto-native-execution/presto_cpp/main/types/tests/PlanConverterTest.cpp [hunk 1]
```diff
@@ -93,7 +94,17 @@ class PlanConverterTest : public ::testing::Test {};
 // Scan + Partial Agg + Repartitioning
 TEST_F(PlanConverterTest, scanAgg) {
   protocol::registerConnector("hive", "hive");
-  assertToVeloxQueryPlan("ScanAgg.json");
+  auto partitionedOutput = assertToVeloxQueryPlan("ScanAgg.json");
+  auto* tableScan = dynamic_cast<const core::TableScanNode*>(
+      partitionedOutput->sources()[0]->sources()[0]->sources()[0].get());
+  ASSERT_TRUE(tableScan);
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/tests/PlanConverterTest.cpp:100]: ASSERT_TRUE(tableScan != nullptr) ? Thanks!

File: presto-native-execution/presto_cpp/main/types/tests/PlanConverterTest.cpp [hunk 2]
```diff
@@ -93,7 +94,17 @@ class PlanConverterTest : public ::testing::Test {};
...
+  auto* columnHandle = dynamic_cast<const connector::hive::HiveColumnHandle*>(
+      tableScan->assignments().at("complex_type").get());
+  ASSERT_TRUE(columnHandle);
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/tests/PlanConverterTest.cpp:103]: diitto

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 3]
```diff
@@ -90,14 +90,24 @@ connector::hive::HiveColumnHandle::ColumnType toHiveColumnType(
   }
 }
 
+std::vector<common::Subfield> toRequiredSubfields(
+    const protocol::List<protocol::Subfield>& subfields) {
+  std::vector<common::Subfield> result;
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:95]:
```
result.reserve(subfields.size());
```

**Review by xiaoxmeng — APPROVED** [review]
@Yuhta Some nits. Thanks!

**Pull request merged**